app/validators/unreserved_username_validator.rb

```diff
@@ -11,16 +11,31 @@ def validate(account)
 
   private
 
+  def reserved_username?
+    pam_username_reserved? || settings_username_reserved?
+  end
+
+  def pam_username_reserved?
+    pam_controlled? && pam_reserves_username?
+  end
+
   def pam_controlled?
-    return false unless Devise.pam_authentication && Devise.pam_controlled_service
+    Devise.pam_authentication && Devise.pam_controlled_service
+  end
 
-    Rpam2.account(Devise.pam_controlled_service, @username).present?
+  def pam_reserves_username?
+    Rpam2.account(Devise.pam_controlled_service, @username)
```

Comment:
Why drop the `.present?` here?

```suggestion
    Rpam2.account(Devise.pam_controlled_service, @username).present?
```

Reply:
I believe `Rpam2.account` always returns true/false, and not account/nil/etc.